spec/models/grant_spec.rb

```diff
@@ -74,7 +72,7 @@
         end
       end
       it "does not allow filtering by agency" do
-        grants = Grant.list(worker1, agency_id: agency2.id)
+        grants = Grant.list(worker1, agency_id: grant3.user.agency.id)
         expect(grants).to_not include(grant3)
       end
     end
```

Comment:
FYI - I did some refactoring on this test.  Added some more unit tests.  I can integrate when my stuff goes in.